ui/src/environments/environmentLoader.ts

```diff
@@ -0,0 +1,20 @@
+import { environment as defaultEnvironment } from './environment';
+
+export const environmentLoader = new Promise<any>( (resolve) => {
+
+  let xmlhttp = new XMLHttpRequest(),
+    method = 'GET',
+    url = './assets/environments/environment.json';
+
+  xmlhttp.open(method, url, true);
+
+  xmlhttp.onload = function() {
```

Comment:
This should handle `onerror`

Reply:
Makes sense to me, I will try to add error handling for both the `xmlhttp.open()` and this loading process.